engine/gamesys/src/gamesys/resources/res_sound_data.cpp

```diff
@@ -41,6 +98,204 @@ namespace dmGameSystem
         return type;
     }
 
+    static dmResource::Result DestroyResource(SoundDataResource* resource)
+    {
+        // The sound data is reference counted
+        // The references are held here and also by any playing voices (e.g. like a long playing music track)
+        dmSound::SetSoundDataCallback(resource->m_SoundData, 0, 0);
+        dmSound::Result r = dmSound::DeleteSoundData(resource->m_SoundData);
+
+        free((void*)resource->m_Path);
+        delete resource;
+
+        return dmSound::RESULT_OK == r ? dmResource::RESULT_OK : dmResource::RESULT_INVAL;
+    }
+
+    static SoundDataChunk* AllocChunk(uint8_t* data, uint32_t data_size, uint32_t offset)
+    {
+        SoundDataChunk* chunk = new SoundDataChunk;
+        chunk->m_Size   = data_size;
+        chunk->m_Offset = offset;
+        chunk->m_Data = new uint8_t[data_size];
+        memcpy(chunk->m_Data, data, data_size);
+        return chunk;
+    }
```

Comment:
It's convenient to have all sound chunks the same size, to help with prefetching data.